proxy/src/main/java/com/velocitypowered/proxy/scheduler/VelocityScheduler.java

```diff
@@ -187,6 +187,7 @@ public void run() {
         } catch (Exception e) {
           Log.logger.error("Exception in task {} by plugin {}", runnable, plugin, e);
         } finally {
+          onFinish();
```

Comment:
i think you should call onFinish() only if task is **not** repeatable

Reply:
You're absolute right, I missed it :)